src/quicktimevideo.cpp

```diff
@@ -0,0 +1,1622 @@
+  @return Returns true, if Tag is found in the ignoreList[]
+ */
+bool dataIgnoreList(Exiv2::DataBuf& buf) {
+  const char ignoreList[8][5] = {
+      "moov", "mdia", "minf", "dinf", "alis", "stbl", "cmov", "meta",
+  };
+
+  for (int i = 0; i < 8; ++i)
+    if (equalsQTimeTag(buf, ignoreList[i]))
+      return true;
+
+  return false;
+}
+
+/*!
+  @brief Function used to convert buffer data into 64-bit
+      signed integer, information stored in Big Endian format
+  @param buf Data buffer that will contain data to be converted
+  @return Returns a signed 64-bit integer
+ */
+int64_t returnBufValue(Exiv2::DataBuf& buf, int n = 4) {
+  int64_t temp = 0;
+  for (int i = n - 1; i >= 0; i--)
+#ifdef _MSC_VER
+    temp = temp + static_cast<int64_t>(buf.data()[i] * (pow(static_cast<float>(256), n - i - 1)));
+#else
+    temp = temp + buf.data()[i] * (pow((float)256, n - i - 1));
+#endif
+
+  return temp;
+}
+
+/*!
+  @brief Function used to convert buffer data into 64-bit
+      unsigned integer, information stored in Big Endian format
+  @param buf Data buffer that will contain data to be converted
+  @return Returns an unsigned 64-bit integer
+ */
+uint64_t returnUnsignedBufValue(Exiv2::DataBuf& buf, int n = 4) {
+  uint64_t temp = 0;
+  for (int i = n - 1; i >= 0; i--)
+#if _MSC_VER
+    temp = temp + static_cast<uint64_t>(buf.data()[i] * (pow(static_cast<float>(256), n - i - 1)));
+#else
+    temp = temp + buf.data()[i] * (pow((float)256, n - i - 1));
+#endif
+
+  return temp;
+}
+
+}  // namespace Internal
+}  // namespace Exiv2
+
+namespace Exiv2 {
+
+using namespace Exiv2::Internal;
+
+QuickTimeVideo::QuickTimeVideo(BasicIo::UniquePtr io) : Image(ImageType::qtime, mdNone, std::move(io)), timeScale_(1) {
+}  // QuickTimeVideo::QuickTimeVideo
+
+std::string QuickTimeVideo::mimeType() const {
+  return "video/quicktime";
+}
+
+void QuickTimeVideo::writeMetadata() {
+}
+
+void QuickTimeVideo::readMetadata() {
+  if (io_->open() != 0)
+    throw Error(ErrorCode::kerDataSourceOpenFailed, io_->path(), strError());
+
+  // Ensure that this is the correct image type
+  if (!isQTimeType(*io_, false)) {
+    if (io_->error() || io_->eof())
+      throw Error(ErrorCode::kerFailedToReadImageData);
+    throw Error(ErrorCode::kerNotAnImage, "QuickTime");
+  }
+
+  IoCloser closer(*io_);
+  clearMetadata();
+  continueTraversing_ = true;
+  height_ = width_ = 1;
+
+  xmpData_["Xmp.video.FileSize"] = (double)io_->size() / (double)1048576;
+  xmpData_["Xmp.video.MimeType"] = mimeType();
+
+  while (continueTraversing_)
+    decodeBlock();
+
+  aspectRatio();
+}  // QuickTimeVideo::readMetadata
+
+void QuickTimeVideo::decodeBlock() {
+  const long bufMinSize = 4;
+  DataBuf buf(bufMinSize + 1);
+  uint64_t size = 0;
+  buf.data()[4] = '\0';
+
+  io_->read(buf.data(), 4);
+  if (io_->eof()) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  size = Exiv2::getULong(buf.data(), bigEndian);
+
+  io_->read(buf.data(), 4);
```

Comment:
```suggestion
  io_->readOrThrow(buf.data(), 4, Exiv2::ErrorCode::kerCorruptedMetadata);
```

I think it would be a good idea to do the same replacement throughout this file, to avoid bugs caused by partial reads. I think the only exception is line 576, where there's already a check for `eof`. Similarly, replace `seek` with `seekOrThrow`.